### PR TITLE
Addition of a CI tool to detect dead links in test/config.json

### PR DESCRIPTION
**Pull request by frouioui**
## Description

This Pull Request fixes the issue detailed in https://github.com/vitessio/vitess/issues/11667. It removes the tests in `test/config.json` that no longer exists and adds a go tool to automatically check that there is no dead links. This tool has been added to the CI workflow `Static Code Checks Etc` to automatically prevent the merge of dead links in `test/config.json`.

We most likely need to do a manual backport on all branches.

## Related Issue(s)

- Fixes https://github.com/vitessio/vitess/issues/11667

## Checklist

-   [x] "Backport me!" label has been added if this change should be backported
-   [x] Tests were added or are not required
-   [x] Documentation was added or is not required

### CODE REVIEW

**Review by GuptaManan100 — APPROVED** [review]
I like the change.

**Review by systay — APPROVED** [review]

File: test/config.json
Comment (deepthi) [inline test/config.json:723]: The reason this became a dead link is that the test was moved from `derived` to `queries` and then the directory was deleted.
However, the test has not been added to config.json with the new path, which means that this test has actually been lost from CI. It could be broken and we won't know.
PR where original change happened: #11379 
I would have expected that this test failed on that PR before it was merged, but I haven't checked.
In any case, instead of deleting this, we should be renaming it to vtgate_queries_derived and fixing the path.

**Review by deepthi — DISMISSED** [review]

File: test/config.json
Comment (frouioui) [reply test/config.json:723]: That's a good catch! I'll fix that.

However, I will leave the tool as it is, it simply reports what are the dead links, then it is up to the user to delete or rename the cluster path.

Comment (deepthi) [reply test/config.json:723]: Agreed, No changes to the tool, just to config.json.
Once that is done, you may dismiss my review and merge this PR.

**Comment by rsajwani** [comment]
Still some test failures are there which seems related to this PR..

**Comment by deepthi** [comment]
> Still some test failures are there which seems related to this PR..

It seems to me that the `derived` test has not been running for some time in CI and is possibly broken for real.
cc @systay

**Comment by frouioui** [comment]
> It seems to me that the derived test has not been running for some time in CI and is possibly broken for real.

Should be fixed now. The clean up code was missing a table, a test was flaky, and a test is now skipped and will be fixed later (https://github.com/vitessio/vitess/issues/11703).

**Review by mattlord — APPROVED** [review]
❤️

**Review by deepthi — APPROVED** [review]

**Pull request merged**
